addons/product/migrations/13.0.1.2/post-migration.py

```diff
@@ -17,6 +17,19 @@ def fill_product_template_attribute_value_attribute_line_id(env):
             ptav.product_attribute_value_id = avtalr.product_attribute_value_id
         """,
     )
+    # TODO: add comment
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE product_template_attribute_value ptav
+        SET attribute_line_id = ptal.id
+        FROM product_template_attribute_line ptal
+        JOIN product_attribute_value pav ON pav.attribute_id = ptal.attribute_id
+        JOIN product_template pt ON ptal.product_tmpl_id = pt.id
```

Comment:
Here you are finding the ptal through pav using attribute_id. But the correct link between pav and ptal is product_attribute_value_product_template_attribute_line_rel table. In https://github.com/OCA/OpenUpgrade/pull/2672, I fill that m2m table in https://github.com/OCA/OpenUpgrade/pull/2672/commits/ed0a6d48a835fb7498cb7c1eafffe3768680553e for archived ptals (the active ones should already have records there). Then, the filling of attribune_line_id (which your PR is about), I move from post-migration to pre-migration in https://github.com/OCA/OpenUpgrade/pull/2672/commits/398cea93e36983fdb0757b9db5d3ea0f13a40ddc and https://github.com/OCA/OpenUpgrade/pull/2672/commits/f58f94c5aeba7da742c7f46d5a0a1127c3e03931. If those two commits don't fix your issue, then please check what thing about these two commits are wrong.

Reply:
The problem here https://github.com/OCA/OpenUpgrade/commit/ed0a6d48a835fb7498cb7c1eafffe3768680553e:
![изображение](https://user-images.githubusercontent.com/33932458/139908452-b44f52ad-4f95-401d-bbda-ea6c0767dd7f.png)
`insert_ptal` not contain lines for my case.
Because that query needs only for case when we remove full line like:
![изображение](https://user-images.githubusercontent.com/33932458/139909846-7cc12366-b242-4005-b787-ea57f7b727ad.png)
![изображение](https://user-images.githubusercontent.com/33932458/139910140-eb7bb7db-cc6e-4420-a93b-2b8bce10da2c.png)

But I have a case when not removing all values from the attribute line:
![изображение](https://user-images.githubusercontent.com/33932458/139910458-95086cfb-90dc-4efe-8a3b-67926b65e786.png)

Reply:
Interesting, I will check! :D

Reply:
I can attach DB created for all of these cases if you need

Reply:
Ok, if it's not a problem for you, attach the DB :D

Reply:
[test_product_v12.zip](https://github.com/OCA/OpenUpgrade/files/7468429/test_product_v12.zip)
login: `admin`
password: `admin`